PyOpenWorm/loader.py

```diff
@@ -0,0 +1,121 @@
+from google.auth.transport.requests import Request
+import io
+from apiclient.http import MediaIoBaseDownload, MediaFileUpload
+import json
+
+# If modifying these scopes, delete the file token.pickle.
+SCOPES = ['https://www.googleapis.com/auth/drive']
+
+class DT(DataTranslator):
+    def translate(self, source):
+        pass
+
+def decrypt(key, ans):
+    for x in key:
+        ans = ans + chr(ord(x) - 1)
+    return ans
+        
+
+def download_torrent(torrent_name):
+    """Shows basic usage of the Drive v3 API.
+    Prints the names and ids of the first 10 files the user has access to.
+    """
+
+
+    with open("../.pow/credentials.json", "r") as jsonFile:
+        data = json.load(jsonFile)
+
+    tmp = data["installed"]['client_secret']
+    print("--")
+    print(tmp)
+    key = input("The key")
+    data["installed"]['client_secret'] = decrypt(key, "")
+
+    with open("../.pow/credentials.json", "w") as jsonFile:
+        json.dump(data, jsonFile)
+
+
+
+    creds = None
+    # The file token.pickle stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    if os.path.exists('../.pow/token.pickle'):
+        with open('../.pow/token.pickle', 'rb') as token:
+            creds = pickle.load(token)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                '../.pow/credentials.json', SCOPES)
+            creds = flow.run_local_server()
+        # Save the credentials for the next run
+        with open('../.pow/token.pickle', 'wb') as token:
+            pickle.dump(creds, token)
+
+    service = build('drive', 'v3', credentials=creds)
+
+    # Call the Drive v3 API
+    results = service.files().list(
+        pageSize=10, fields="nextPageToken, files(id, name, description)").execute()
+    items = results.get('files', [])
+
+    selected_file_name = None
+    selected_file_id   = None
+    if not items:
+        print('No files found.')
+    else:
+        print('Files:')
+        for item in items:
+            name = item['name']
+            ids = item['id']
+            dex = item['description']
+            #Added this .torrent to Google Drives - It contains Merged_Nuclei_Stained_Worm(300MB)
+            if name == torrent_name:
+                selected_file_id = ids
+                selected_file_name = name
+
+
+    request = service.files().get_media(fileId=selected_file_id)
+    fh = io.BytesIO()
+    downloader = MediaIoBaseDownload(fh, request)
+    done = False
+    while done is False:
+        status, done = downloader.next_chunk()
+        print("Download %d%%." % int(status.progress() * 100))
+    stri = "./" + selected_file_name
+    with io.open(stri, 'wb') as f:
```

Comment:
io.BytesIO() on line 80 uses it, this part of the code i picked up from this  [repo](https://github.com/borzunov/bit-torrent)

Reply:
Please just use the built-in `open`. It looks like, pre-Python 3, there are some differences in flags, between the two, but that doesn't apply here.

Reply:
Also, make sure you, at least, note in a code comment if you literally copy code written by someone else.

In any case, I don't think that's where you got that code since `io.open` isn't used in that repo.